aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cpp

```diff
@@ -482,15 +627,34 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda_lib(const Tensor& self, bool
     VT_working_copy.transpose_(-2, -1);
   }
 
+  // VT_working_copy is row-major, actually "V" now
+
   // heuristic for using `gesvdjBatched` over `gesvdj`
   if (m <= 32 && n <= 32 && batch_size > 1 && (!some || m == n)) {
     apply_svd_lib_gesvdjBatched(self, U_working_copy, S_working_copy, VT_working_copy, infos, compute_uv);
+
+    // A device-host sync will be performed.
+    batchCheckErrors(infos, "svd_cuda");
   } else {
     apply_svd_lib_gesvdj(self, U_working_copy, S_working_copy, VT_working_copy, infos, compute_uv, some);
-  }
 
-  // A device-host sync will be performed.
-  batchCheckErrors(infos, "svd_cuda");
+    // A device-host sync will be performed.
+    std::vector<int64_t> gesvdj_convergence_check = _check_gesvdj_convergence(infos, std::min(m, n) + 1);
+
+    if (!gesvdj_convergence_check.empty()) {
+      // fall back to gesvd path for those non-converging batches
+
+      TORCH_WARN_ONCE("cuSOLVER SVD failed to converge for the given inputs. "
+                      "A more accurate method will be used to calculate the SVD as a fallback.");
```

Comment:
Perhaps report also the number of inputs for which it failed and the number of the first input that failed so that the user can inspect what went wrong?